src/components/NewMovie/NewMovie.js

```diff
@@ -1,13 +1,112 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './NewMovie.scss';
 
 export class NewMovie extends Component {
-  state = {};
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  handleAdd = (event) => {
+    event.preventDefault();
+    const { title, description, imdbId, imdbUrl, imgUrl } = this.state;
+
+    if (Object.values(this.state).some(item => !item)) {
+      return;
+    }
+
+    this.props.onAdd({
+      title,
+      description,
+      imdbId,
+      imdbUrl,
+      imgUrl,
+    });
+
+    this.setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  }
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value.trimLeft(),
+    });
+  }
```

Comment:
Можна використати деструктуризацію і трошки скоротити запис:  
```suggestion
  handleChange = (event) => {
      const { name, value } = event.target;
      this.setState({ [name]: value.trimLeft() });
    }
```